set.mm

```diff
@@ -472151,6 +472186,240 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
+      ( vy wcel cv co wceq wrex csn cxp cima snssd elgrpsumset wb eqeq2d rexsng
+      oveq2 syl rexbidv bitrd ) AIFCHUAZUBUCPIBQZOQZERZSZOUMTZBCTIUNHERZSZBCTAB
+      ODEFCUMGIJKLMAHDNUDUEAURUTBCAHDPURUTUFNUQUTOHDUOHSUPUSIUOHUNEUIUGUHUJUKUL
+      $.
+  $}
+
+  ${
+    mndsumsetss.1 $e |- B = ( Base ` G ) $.
+    mndsumsetss.2 $e |- .+^ = ( +f ` G ) $.
+    mndsumsetss.3 $e |- ( ph -> G e. Mnd ) $.
+    mndsumsetss.4 $e |- ( ph -> E C_ B ) $.
+    mndsumsetss.5 $e |- ( ph -> F C_ B ) $.
+    $( Closure for the sumset operation in a monoid.  (Contributed by Thierry
+       Arnoux, 20-Jan-2024.) $)
+    mndsumsetss $p |- ( ph -> ( .+^ " ( E X. F ) ) C_ B ) $=
+      ( cmnd wcel cxp wf cima wss mndplusf fimass 3syl ) AFLMBBNZBCOCDENZPBQIBC
+      FGHRUABCUBST $.
+  $}
+
+  ${
+    $d .+ g h x y $.  $d .+^ h k $.  $d .~ h k $.  $d A g h k x y $.
+    $d B g h k x y $.  $d X g h k x y $.  $d Y h $.  $d Y x y $.  $d h k ph $.
+    sumsetsnorb.1 $e |- B = ( Base ` G ) $.
+    sumsetsnorb.2 $e |- .+ = ( +g ` G ) $.
+    sumsetsnorb.3 $e |- .+^ = ( +f ` G ) $.
+    sumsetsnorb.4 $e |- .~ = { <. x , y >. | ( { x , y } C_ B
+                                             /\ E. g e. A ( g .+ x ) = y ) } $.
+    sumsetsnorb.5 $e |- ( ph -> G e. Mnd ) $.
+    sumsetsnorb.6 $e |- ( ph -> A C_ B ) $.
+    sumsetsnorb.7 $e |- ( ph -> X e. B ) $.
+    $( The sumset of a group with a single element is the element's orbit by
+       the group action.  See ~ gaorb .  (Contributed by Thierry Arnoux,
+       21-Jan-2024.) $)
+    sumsetsnorb $p |- ( ph -> ( .+^ " ( A X. { X } ) ) = [ X ] .~ ) $=
+      ( vh wcel vk csn cxp cima cec cv snssd mndsumsetss sselda wss crn imassrn
+      df-ec cpr co wceq wrex copab rneqi wex cab rnopab vex prss biimpri simprd
+      wa adantr exlimiv abssi eqsstri sstri a1i wbr w3a anim1i biantrurd df-3an
+      eqsstrid syl6bbr gaorb syl6rbbr cvv wb elecg sylancr elgrpsumsetsn rexbii
+      eqcom syl6bb 3bitr4rd eqrdav ) AUAGDKUBZUCUDZKHUEZEAWNEUAUFZAEGDWMJLNPQAK
+      ERUGUHUIAWOEWPAWOHWMUDZEKHUMWQEUJAWQHUKZEHWMULWRBUFZCUFZUNEUJZIUFWSFUOWTU
+      PIDUQZVGZBCURZUKZEHXDOUSXEXCBUTZCVAEXCBCVBXFCEXCWTETZBXAXGXBXAWSETZXGXHXG
+      VGXAWSWTEBVCCVCVDVEVFVHVIVJVKVKVLVMVSUIAWPETZVGZKWPHVNZSUFKFUOZWPUPZSDUQZ
+      WPWOTZWPWNTZXJXNKETZXIXNVOZXKXJXNXQXIVGZXNVGXRXJXSXNAXQXIRVPVQXQXIXNVRVTB
+      CKWPFHISDEOWAWBXJWPWCTXQXOXKWDUAVCAXQXIRVHZWPKHWCEWEWFXJXPWPXLUPZSDUQXNXJ
+      SDEFGJKWPLMNADEUJXIQVHXTWGYAXMSDWPXLWIWHWJWKWL $.
+  $}
+
+  ${
+    $d .X. x y $.  $d .x. x y $.  $d B x y $.  $d E x y $.  $d F x y $.
+    $d R x y $.  $d Z x y $.  $d ph x y $.
+    elringsumset.1 $e |- B = ( Base ` R ) $.
+    elringsumset.2 $e |- .x. = ( .r ` R ) $.
+    elringsumset.3 $e |- G = ( mulGrp ` R ) $.
+    elringsumset.4 $e |- .X. = ( +f ` G ) $.
+    elringsumset.5 $e |- ( ph -> E C_ B ) $.
+    elringsumset.6 $e |- ( ph -> F C_ B ) $.
+    $( Membership to a sumset for a ring operation.  (Contributed by Thierry
```

Comment:
I think the usage of the term "sumset" here is a bit confusing, since `( .X. " ( E X. F ) )` is actually a product set. I think it will more clear to spell things out as "Membership in a product of two subsets of a ring".